### PR TITLE
z_en_item00

### PR DESCRIPTION
**Pull request by Rozelette**
17/21 functions OK, with the others pretty close to it, but as I have not been making headway with them I am PRing it. I added a decent amount of documentation formatted the same way as OOT. I could not find a definitive source of info for MM's item IDs, so I only filled in what I could absolutely determine. I also threw in a quick way to split rodata for code files, but I didn't test it out extensively.

### CODE REVIEW

**Review by Kenix3 — CHANGES_REQUESTED** [review]
Looks good to me. Just have a few naming issues.

Also, looks like you have a merge conflict. Please update your PR to rebase on master. Thank you for the submission!

File: include/macros.h [hunk 1]
```diff
@@ -19,10 +19,14 @@
...
 #define SET_NEXT_GAMESTATE(curState, newInit, newStruct) \
             (curState)->nextGameStateInit = (GameStateFunc)newInit; \
-            (curState)->nextGameStateSize = sizeof(newStruct); 
-        
+            (curState)->nextGameStateSize = sizeof(newStruct);
+
 #define PLAYER ((ActorPlayer*)globalCtx->actorCtx.actorList[ACTORCAT_PLAYER].first)
 
+// TODO do these names still make sense in MM?
+#define LINK_IS_CHILD (gSaveContext.perm.linkAge != 0)
```
[the remark this answers is not in the document]
Comment (Rozelette) [reply include/macros.h:27]: I think it's just a remnant from OOT. As far as I can tell looking at it in Ghidra, it is never set to anything other than 0. However, it is checked at various places in the code. It is also still called `link_age` internally. Would you prefer these macros and the variable rename to `unk7`?

File: src/code/z_en_item00.c [hunk 2]
```diff
@@ -0,0 +1,1167 @@
...
+    ITEM00_MAGIC_SMALL, ITEM00_MAGIC_SMALL, ITEM00_MAGIC_SMALL, ITEM00_MAGIC_SMALL, ITEM00_MAGIC_SMALL, ITEM00_MAGIC_SMALL, ITEM00_MAGIC_SMALL, ITEM00_MAGIC_SMALL, ITEM00_MAGIC_SMALL, ITEM00_MAGIC_SMALL, ITEM00_MAGIC_LARGE, ITEM00_MAGIC_LARGE, ITEM00_MAGIC_LARGE, ITEM00_MAGIC_LARGE, ITEM00_MAGIC_LARGE, ITEM00_MAGIC_LARGE,
+    ITEM00_NO_DROP, ITEM00_NO_DROP, ITEM00_NO_DROP, ITEM00_NO_DROP, ITEM00_NUTS_1, ITEM00_NUTS_1, ITEM00_NO_DROP, ITEM00_ARROWS_10, ITEM00_ARROWS_10, ITEM00_ARROWS_10, ITEM00_STICK, ITEM00_STICK, ITEM00_NO_DROP, ITEM00_HEART, ITEM00_HEART, ITEM00_FLEXIBLE,
+    ITEM00_RUPEE_GREEN, ITEM00_RUPEE_BLUE, ITEM00_RUPEE_BLUE, ITEM00_RUPEE_RED, ITEM00_NO_DROP, ITEM00_ARROWS_10, ITEM00_ARROWS_10, ITEM00_ARROWS_30, ITEM00_BOMBS_A, ITEM00_NO_DROP, ITEM00_STICK, ITEM00_MAGIC_SMALL, ITEM00_MAGIC_LARGE, ITEM00_HEART, ITEM00_HEART, ITEM00_FLEXIBLE,
+    ITEM00_RUPEE_GREEN, ITEM00_HEART, ITEM00_HEART, ITEM00_MAGIC_SMALL, ITEM00_MASK, ITEM00_NO_DROP, ITEM00_NO_DROP, ITEM00_NO_DROP, ITEM00_NO_DROP, ITEM00_NO_DROP, ITEM00_NO_DROP, ITEM00_NO_DROP, ITEM00_NO_DROP, ITEM00_NO_DROP, ITEM00_NO_DROP, ITEM00_NO_DROP,
+};
+
+u8 sDropTableAmounts[16 * 17] = {
```
[the remark this answers is not in the document]
Comment (Rozelette) [reply src/code/z_en_item00.c:970]: 17 drop tables of 16 entries each. I was somewhat unsure making them a macro but I'll do so

**Review by Kenix3 — APPROVED** [review]
We decided to go ahead and keep linkAge, since it's not used, and it's in MM debug screen strings.
Ready to merge as soon as conflicts are resolved.

**Pull request merged**